static/packages/autocomplete.js

```diff
@@ -17,124 +17,97 @@ class AutocompleteField {
         const place = self.autocomplete.getPlace();
         let address = new Address(place, self.input.value);
 
-        self.input.value = address.fields['formatted'];
-        self.city.value = address.fields['city'];
-        self.street.value = address.fields['street'];
-        self.streetNumber.value = address.fields['street_number'];
+        self.input.value = address.formatted;
+        self.city.value = address.city;
+        self.street.value = address.street;
+        self.streetNumber.value = address.street_number;
     }
 }
 
 class Address {
     constructor(place, input) {
         if (typeof place === 'undefined' || typeof place.address_components === 'undefined')
-            this.fields = this.customAddressFormatting(this, input);
+            this.customAddressFormatting(this, input);
         else
-            this.fields = this.extractAddressComponents(place);
+            this.extractAddressComponents(this, place);
     }
 
     customAddressFormatting(self, customInput) {

```

Comment:
This still doesn't seem to work that well. Let me suggest something:

Your regex usage is far from the potential available here. Check the following code:

```javascript
let customInput = 'Haifa st 5, Tel Aviv'
let match = customInput.match(/(\w+\s?\w*)\s*(\d*)\s*,\s*(\w+\s?\w*)\s*/);

let street = match[1];
let number = match[2];
let city = match[3];
```
This will result with:
- `street="Haifa st"`
- `number="5"`
- `city="Tel Aviv"`

This is the true power of regex. It can work for more inputs:
- `Haifa st, Tel Aviv`
   - `street="Haifa st"`
   - `number=""`
   - `city="Tel Aviv"`

This regex matches the full format. You can also utilize multiple _different_ regex formats for different inputs. For example: the input `"Tel Aviv"` will not match the regex above. So if `match` is `None`, we can try a different regex format.

Checkout:
- [Playground for regex](https://regexr.com/)
- [Javascript regex groups](https://javascript.info/regexp-groups)
- [regex cheatsheet](https://www.rexegg.com/regex-quickstart.html)

Reply:
Listen, I didn't really have time to go over all these stupid symbols. Why regex is so convoluted ? isn't there an api where I could just build up like `regex.thenWord().thenDigit()` or whatever? I think I'll add a better use of regex like you described here in the next PR

Reply:
This comment is more about how `customAddressFormatting` doesn't actually function that well. I was merely offering a fix. So you really should fix it now.

Regex is complex, but that also allows it to be pretty powerful. Its also why there are [sites](https://regexr.com/) to help with creating and understanding it. I'm not familiar with an API that simplifies it. 

Look, you can use the following snippet:

```javascript
let match = customInput.match(/^(\w+\s?\w*)\s*(?:(\d*)\s*)?(?:,\s*(\w+\s?\w*)?\s*)?$/);
if (match == null) {
	// bad input!, alert the user, or comeup with a regex to support the input
  return;
}

if (match[2] == null && match[3] == null) {
	// single token match. Something like:
  // Tel Aviv
  // Haifa
  let city = match[1];
} else if (match[3] == null) {
	// 2 token match. Something like:
  // Balkan 2
  // Halbok 10
  let streetName = match[1];
  let streetNumber = match[2];
} else if (match[2] == null) {
	// Second kind of 2 token match. Something like:
  // Brioli, Tel Aviv
  let streetName = match[1];
  let city = match[3];
} else {
	// 3 token match. Something like:
  // Brok 10, Tel Aviv
  // Lolipop 5, Haifa
  let street = match[1];
	let number = match[2];
	let city = match[3];
}
```

Reply:
ok ;-; didnt implement the null varient tho

Reply:
Might be important